File: be_camphub/src/main/java/org/camphub/be_camphub/dto/response/cart/CartItemResponse.java
package org.camphub.be_camphub.dto.response.cart;

import java.math.BigDecimal;
import java.util.UUID;

import lombok.*;
import lombok.experimental.FieldDefaults;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CartItemResponse {
    UUID id;
    UUID cartId;
    UUID itemId;

    // Item details
    String itemName;
    // image URL of the item
    String itemImage;

    BigDecimal depositAmount;

    Integer quantity;
    Integer rentalDays;
    BigDecimal price;
    BigDecimal subtotal;

    Boolean isAvailable; // if quantity are available
}

File: fe_camphub/src/libs/core/dto/response/cart-item.response.ts
export interface CartItemResponse {
    id: string;         // UUID
    cartId: string;     // UUID
    itemId: string;     // UUID

    itemName: string;
    itemImage: string;

    depositAmount: number;
    
    quantity: number;
    rentalDays: number;
    price: number;      // BigDecimal -> number
    subtotal: number;   // BigDecimal -> number

    isAvailable: boolean;
}
